specification/advisor/resource-manager/Microsoft.Advisor/stable/2022-10-01/advisor.json

```diff
@@ -0,0 +1,1762 @@
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. Successfully retrieved zero or more configurations.",
+            "schema": {
+              "$ref": "#/definitions/ConfigurationListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ArmErrorResponse"
+            }
+          }
+        },
+        "deprecated": false,
+        "x-ms-pageable": {
+          "nextLinkName": null
+        },
+        "x-ms-examples": {
+          "GetConfigurations": {
+            "$ref": "./examples/ListConfigurations.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.Advisor/configurations/{configurationName}": {
+      "put": {
+        "tags": [
+          "Configurations"
+        ],
+        "summary": "Create/Overwrite Azure Advisor configuration.",
+        "operationId": "Configurations_CreateInResourceGroup",
+        "parameters": [
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "name": "configContract",
+            "in": "body",
+            "description": "The Azure Advisor configuration data structure.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ConfigData"
+            }
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "name": "configurationName",
+            "in": "path",
+            "description": "Advisor configuration name. Value must be 'default'",
+            "required": true,
+            "type": "string",
+            "enum": [
+              "default"
+            ],
+            "x-ms-enum": {
+              "name": "ConfigurationName",
+              "modelAsString": true
+            }
+          },
+          {
+            "name": "resourceGroup",
+            "in": "path",
+            "description": "The name of the Azure resource group.",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully created/overwrote configuration.",
+            "schema": {
+              "$ref": "#/definitions/ConfigData"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ArmErrorResponse"
+            }
+          }
+        },
+        "deprecated": false,
+        "x-ms-examples": {
+          "PutConfigurations": {
+            "$ref": "./examples/CreateConfiguration.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Advisor/generateRecommendations": {
+      "post": {
```

Comment:
Since you are returning 202, I presume this is a long running operation. You must have x-ms-long-running-operation and x-ms-long-running-operation-options extensions.

[https://github.com/Azure/azure-openapi-validator/blob/main/docs/long-running-operations-options-validator.md](https://github.com/Azure/azure-openapi-validator/blob/main/docs/long-running-operations-options-validator.md)

[https://github.com/Azure/azure-openapi-validator/blob/main/docs/lro-extension.md](https://github.com/Azure/azure-openapi-validator/blob/main/docs/lro-extension.md)

Reply:
Thanks for the review - Advisor team has this comment taken as PBI to include the change suggested here : https://msazure.visualstudio.com/One/_backlogs/backlog/AzureAdvisor/Features/?workitem=17048171 and will introduce the change needed by the comment in the next api-version increment of advisor.